### PR TITLE
Rename TOTAL_MEMORY to INITIAL_MEMORY and WASM_MEM_MAX to MAXIMUM_MEMORY

### PR DESCRIPTION
**Pull request by kripken**
As suggested (first half) by @dschuff . The new names are consistent with
wasm's names for them, and more accurate (as TOTAL_MEMORY wasn't
necessarily the total memory size if growth was enabled).

The old names are still supported as aliases.

### CODE REVIEW

File: site/source/docs/api_reference/emscripten.h.rst [hunk 1]
```diff
@@ -92,13 +92,13 @@ Defines
...
-    EM_JS(int, get_total_memory, (), {
-      return TOTAL_MEMORY;
+    EM_JS(int, get_INITIAL_MEMORY, (), {
+      return INITIAL_MEMORY;
     });
 
     int main() {
       int x = add_forty_two(100);
-      int y = get_total_memory();
+      int y = get_INITIAL_MEMORY();
```
Comment (sbc100) [inline site/source/docs/api_reference/emscripten.h.rst]: Maybe leave the function name alone?

File: tests/core/test_memorygrowth_3.c [hunk 2]
```diff
@@ -12,15 +12,15 @@
 #include <assert.h>
 #include "emscripten.h"
 
-int get_TOTAL_MEMORY() {
+int get_INITIAL_MEMORY() {
```
Comment (sbc100) [inline tests/core/test_memorygrowth_3.c]: ditto

File: tests/core/test_memorygrowth_memory_growth_step.c [hunk 3]
```diff
@@ -12,42 +12,42 @@
 #include <assert.h>
 #include "emscripten.h"
 
-int get_TOTAL_MEMORY() {
+int get_INITIAL_MEMORY() {
```
Comment (sbc100) [inline tests/core/test_memorygrowth_memory_growth_step.c]: ditto

File: site/source/docs/api_reference/emscripten.h.rst (hunk 1 above)
Comment (kripken) [reply site/source/docs/api_reference/emscripten.h.rst]: Good idea!

**Pull request merged**